### DISCUSSION TITLE
Library component LOB/Enterprise

### DISCUSSION THREAD

**Discussion by franklupo** [microsoft/fluentui-blazor#2419] [Ideas]
I am rewriting an application from MudBlazor to FluentUI. I am missing some components and I am creating them.
For example:
- DataFilter
- TextPassword
- OTP Input
- ...

I know they are not part of this library. I could create a library and publish it on NuGet, but I think it is a waste. You have created separate projects for Icons, Emojii, EntityFramework, what do you think about creating a "LOB/Bussiness/Enterprise" component project for these?

Thank you for your availability

**Comment by vnbaaij** [comment]
I think it would be a good idea to have a sort of 'extensions' project. We have always had the idea that the library itself is not meant to provide everything. That is why we, for example, do not include charts or a rich text editor.

So, extension library is okay, but it can't be something in this repo nor in the Microsoft.FluentUI... namespace. If either of that would be the case, it would also be our responsibility and we can't /don't want to take on this extra load. Being part of the microsoft GitHub org is not free in the sense of time and effort. 

Of course we are fully committed to supporting such a endeavor. Maybe someone can set up a new GitHub organization for this?

**Reply by franklupo** [reply]
Organization what do you mean?

**Reply by vnbaaij** [reply]
An organization on GitHub is a sort of container for repositories. It is akin to a user but with more settings for security etc. See https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/about-organizations

**Reply by franklupo** [reply]
Ok, I know and have organizations. What I meant is to create a new one like FluentUIExtension?

**Reply by vnbaaij** [reply]
Yes. Although I'd put Blazor in the org name as well. Otherwise people might think it is React / Javascript stuff.

**Reply by franklupo** [reply]
ok, you have to decide on a good name and if you want I can open it. I would like you to help me in the initial setup of the project.

**Reply by franklupo** [reply]
sorry for my English

**Reply by vnbaaij** [reply]
> ok, you have to decide on a good name and if you want I can open it. I would like you to help me in the initial setup of the project.

Of course! I'll help where I can. Just not sure if I (or Denis) should be admins in that org. Might pose some conflicts of interest in the future. 

How about 'fluentui-blazor-community' for the organization name? Covers everything and makes clear this is not coming from Microsoft. 
We can (should) then also use that as the base namespace `FluentUI.Blazor.Community` in the solutions/projects we create in the repositories. 

So for your extensions project maybe `FluentUI.Blazor.Communty.Components`? 

> sorry for my English

No worries. Almost none of us are native English speakers

**Reply by franklupo** [reply]
Do we use the Fluent component prefix or change?

**Reply by vnbaaij** [reply]
Hmm. Good question. @dvoituron what do you think?
I'm leaning towards changing it to avoid confusion on where the support should come from.

Maybe `FC... `

**Reply by dvoituron** [reply]
Yes. Good question :-)

Copilot replied this:
> Using the same component prefix for both can be convenient for consistency and ease of use, especially if the extensions are designed to integrate seamlessly with FluentUI-Blazor. However, there are a few considerations to keep in mind:
> 
> 1. **Namespace Clarity**: Using the same prefix might cause confusion if both libraries have components with similar names. Distinct prefixes can help avoid naming conflicts and make it clear which library a component belongs to.
> 2. **User Experience**: If the goal is to provide a cohesive experience where users can easily switch between components from both libraries without worrying about different prefixes, then using the same prefix makes sense.
> 3. **Community Standards**: Look at how other popular Blazor component libraries handle this. Consistency with community practices can make your library more approachable for new users.
> 4. **Documentation and Support**: Ensure that your documentation clearly distinguishes between components, regardless of the prefix used.
> Ultimately, the decision depends on your specific goals and the needs of your users. If you prioritize integration and ease of use, the same prefix might be beneficial. If you want to avoid any potential conflicts and ensure clarity, distinct prefixes could be the way to go.

My conclusion would be, is it possible to find a new prefix starting by "Fluent"? "FluentEx", "FluentX", ... 😁

**Reply by franklupo** [reply]
- FbcButton fluent blazor comunity 
- FcButton fluent comunity 
- FexBitton fluent extension 
- FxButton  no Remember java

**Reply by dvoituron** [reply]
Also possible. 

The single advantage having a prefix starting by `Fluent` is for the intellisense usage.

**Reply by vnbaaij** [reply]
Yup, we should definitely start with `Fluent`

**Reply by franklupo** [reply]
- FluentCx community experience
- FluentCc community component

**Reply by vnbaaij** [reply]
I thought about `FluentCx` as well.. . Fluent Community Extension

**Reply by franklupo** [reply]
In summary:
- Organiztion: FluentUI-Blazor-Community
- First Repository: FluentUI.Blazor.Communty
- Preix: FluentCx 

something is missing?

**Reply by vnbaaij** [reply]
LGTM! (looks good to me!)

**Reply by franklupo** [reply]
@dvoituron what do you think?

**Reply by dvoituron** [reply]
Yes, I think `FluentUI.Blazor.Community` namespace and `FluentCx` prefix are good choice.

Other "details" during the project creation: to include a Unit Tests project and GitHub actions to build and verify these tests :-)

**Comment by franklupo** [comment]
What do you think of the logo?

![logo (2)](https://github.com/user-attachments/assets/8585a05d-528d-4519-832e-01873026f87c)

**Reply by vnbaaij** [reply]
We already have this one for the library. I think we can re-use for the community packages to increase brand recognition. I can send you the SVG file 
![Logo-Light](https://github.com/user-attachments/assets/e75375b1-5131-476f-80a9-caa1a1f61773)

**Reply by franklupo** [reply]
great I thought it was from Microsoft and I couldn't use it

**Reply by franklupo** [reply]
The name of first repository? 
- FluentUI.Blazor.Communty
- Components
- Core

**Reply by dvoituron** [reply]
FluentUI.Blazor.Components

Later you could add Samples, Documentation, etc repos

**Reply by franklupo** [reply]
It seems complicated and unclear to me

`https://github.com/fluentui-blazor-community/FluentUI.Blazor.Components`

**Reply by vnbaaij** [reply]
I think it should be `https://github.com/fluentui-blazor-community/FluentUI.Blazor.Community.Components`

The whole path upwards to the last / doesn't really matter for when you clone locally

**Reply by franklupo** [reply]
Perfect.
Now I create the `FluentUI.Blazor.Community.Components` repository but I need it to be set up like this with src, test, action and everything needed to be productive this way. I need your help.

**Reply by franklupo** [reply]
What role do I assign you in the repository?

**Reply by vnbaaij** [reply]
I'd say contributor or member. Just not administrator

**Reply by franklupo** [reply]
I added you.
You could add: the empty project like this with src,test,action.

best regads

**Reply by franklupo** [reply]
Can I clone this repository as a base for https://github.com/fluentui-blazor-community/FluentUI.Blazor.Community.Components?

It would look something like this

**Reply by vnbaaij** [reply]
Yes, you can copy the structure

**Comment by saint4eva** [comment]
Why not something like CommunityToolkit.FluentUI.Blazor? Maui and WinUI have community toolkits. Maybe you can take a look at how they are doing it.

- CommunityToolkit.FluentUI.Blazor.Core

- CommunityToolkit.FluentUI.Blazor.Extensions (then you can have under Extensions things like - Graphs, Charts, Platforms (platform specific), Buttons, Texts, Icons etc)

**Reply by vnbaaij** [reply]
Don't think CommunityToolkit is a good home for this. It is specific to technology areas like MAUI and WinUI. Fluent UI is not such an area but just a narrow part of the Blazor landscape. (so something like CommunityToolkit.Blazor could be a thing!)

**Comment by f4n0** [comment]
Are these CommunityComponents being developed? I've seen the repo but has no files in it!

**Reply by vnbaaij** [reply]
I'm not sure what the status is. @dvoituron and I will try to help there, but we do not own it. If we would, things could just as well be in this repo.

@franklupo Maybe open up the discussion forum in the community repo as well?

**Reply by franklupo** [reply]
> Non sono sicuro di quale sia la situazione.@dvoiturone cercherò di aiutare lì, ma non ne siamo i proprietari. Se lo fossimo, le cose potrebbero benissimo essere in questo repo.
> 
> @franklupoForse dovresti aprire un forum di discussione anche nel repository della community

sorry I'm very busy at the moment. I can't now
